tests/phpunit/tests/media.php

```diff
@@ -707,6 +707,355 @@ public function test_post_gallery_images() {
 		$this->assertSame( $srcs, $ids1_srcs );
 	}
 
+	/**
+	 * @ticket 43826
+	 * @group blocks
+	 */
+	public function test_block_post_galleries() {
+		// Set up an unattached image.
+		$this->factory->attachment->create_object(
+			array(
+				'file'           => 'test.jpg',
+				'post_parent'    => 0,
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			)
+		);
+
+		$post_id = $this->factory->post->create(
+			array(
+				'post_content' => '<!-- wp:gallery -->',
+			)
+		);
+
+		$galleries = get_post_galleries( $post_id, false );
+
+		$this->assertTrue( is_array( $galleries ) );
+		$this->assertEmpty( $galleries[0]['src'] );
```

Comment:
The error in the tests occurs here.

And this is the culprit in source:
https://github.com/WordPress/wordpress-develop/blob/7dc550a619576f0a7e708086e11f2f1637a011fa/src/wp-includes/media.php#L4850

If this line is moved outside of `if ( ! empty( $src ) ) {}`, then `test_mixed_post_galleries()` and `test_mixed_post_galleries_data()` fail as the `Actual` arrays contain an extra `array( 'src' => array() )` element.

See the only other `assertEmpty()` check on this in the test suite [here](https://github.com/costdev/wordpress-develop/blob/f7d670c767254e58a52a9ae1d4b9921678686458/tests/phpunit/tests/media.php#L582), which tests an empty `[gallery]` shortcode.

This suggests that it should return `array( 'src' => array() )` if no `src` is found.

From the [Docs](https://developer.wordpress.org/reference/functions/get_post_galleries/) on this method:
> Return
> (array) A list of arrays, each containing gallery data and srcs parsed from the expanded shortcode.

So, _should_ it return `array( 'src' => array() )` if no `src` is found?
- If so, then  `test_mixed_post_galleries()` and `test_mixed_post_galleries_data()` need to be updated to account for this extra element.
- If not, then the `$this->assertEmpty( $galleries[0]['src'] );` assertion above should be removed.
- If neither, I'm not sure where the method goes wrong.

Reply:
`parse_blocks()` detects a shortcode as a block with a `null` value for `blockName`.
As shortcodes are handled by the code above these changes, the following guard is now used:
```
if( ! $block['blockName'] ) {
	continue;
}
```